tools/build/third_party/breakpad/dump_syms_pe.cc

```diff
@@ -0,0 +1,389 @@
+    if (section->virtualAddress <= address && section->virtualAddress + section->virtualSize > address) {
+      return section;
+    }
+  }
+  return nullptr;
+}
+
+// buffer needs to be 9 bytes (or larger).
+static const char* getSectionName(const PEFile& file, const PESectionHeader& section, char* buffer) {
+  if (section.name[0] == '/') {
+    char offS[8];
+    std::memcpy(offS, &section.name[1], 7);
+    offS[7] = 0;
+    int off = atoi(offS);
+    if (off > file.strings.size) {
+      return "";
+    }
+    return file.strings.data + off;
+  }
+  std::memcpy(buffer, section.name, 8);
+  buffer[8] = 0;
+  return buffer;
+}
+
+static PESectionHeader* findSectionByName(const PEFile& file, const std::string& name) {
+  char nameBuf[9];
+  for (int i = 0; i < file.header->numSections; i++) {
+    PESectionHeader* section = &file.sections[i];
+    const char* sectionName = getSectionName(file, *section, nameBuf);
+    if (sectionName == name) {
+      return section;
+    }
+  }
+  return nullptr;
+}
+
+static bool initIdentifiers(PEFile* file) {
+  if (!file->optionalHeader->directory[6].address) {
+    fprintf(stderr, "File does not contain a debug section\n");
+    return false;
+  }
+  uint32_t address = file->optionalHeader->directory[6].address;
+  PESectionHeader* section = findSection(*file, address);
+  if (section == nullptr) {
+    fprintf(stderr, "Failed to locate debug info section at %x\n", address);
+    return false;
+  }
+
+  uint32_t offset = address - section->virtualAddress;
+  if (section->virtualSize < offset + sizeof(DebugDirectory)) {
+    fprintf(stderr, "Debug section too small: %d < %d\n", section->virtualSize, offset + sizeof(DebugDirectory));
+    return false;
+  }
+  if (section->offset + offset + sizeof(DebugDirectory) > file->data.size()) {
+    fprintf(stderr, "Debug section past EOF: %x\n", section->offset);
+    return false;
+  }
+  DebugDirectory* directory = (DebugDirectory*)(file->data.data() + section->offset + offset);
+
+  if (directory->type != 2) {
+    fprintf(stderr, "Unsupported debug data type: %x\n", directory->type);
+    return false;
+  }
+  if (directory->size < sizeof(DebugData) + 1 /* 0 length string is the min */) {
+    fprintf(stderr, "Debug data too small: %d\n", directory->size);
+    return false;
+  }
+  if (directory->offset + directory->size > file->data.size()) {
+    fprintf(stderr, "Debug data past EOF: %x\n", directory->offset);
+    return false;
+  }
+  DebugData* debugData = (DebugData*)(file->data.data() + directory->offset);
+
+  char debugIdentifier[32 + 8 + 1];
+  snprintf(debugIdentifier, sizeof(debugIdentifier), "%08X%04X%04X%02X%02X%02X%02X%02X%02X%02X%X\0",
+      debugData->uuid.a, debugData->uuid.b, debugData->uuid.c,
+      debugData->uuid.d[0], debugData->uuid.d[1], debugData->uuid.d[2], debugData->uuid.d[3],
+      debugData->uuid.d[4], debugData->uuid.d[5], debugData->uuid.d[6], debugData->uuid.d[7],
+      debugData->age);
```

Comment:
What is `age` doing here (or is the last %X the age, if so you're missing a byte for the uuid)?